src/hardware/input/mouseif_virtual_machines.cpp

```diff
@@ -62,124 +48,199 @@ union VMwareButtons {
 	bit_view<3, 1> middle;
 };
 
-static constexpr uint32_t VMWARE_MAGIC = 0x564D5868u;  // magic number for all VMware calls
-static constexpr uint32_t ABS_UPDATED  = 4; // tells about new pointer position
-static constexpr uint32_t ABS_NOT_UPDATED = 0;
+static struct {
+	bool is_active = false;
+
+	// true = guest driver wants host mouse pointer visible
+	bool wants_pointer = false;
+
+} virtualbox;
+
+static struct {
+	bool is_active = false;
+
+	bool updated = false;       // true = state update waits to be picked up
+	VMwareButtons buttons = {}; // state of mouse buttons, in VMware format
+	int8_t counter_w      = 0;  // wheel movement counter
+} vmware;
 
 static bool use_relative = true; // true = ignore absolute mouse position, use relative
 static bool is_input_raw = true; // true = no host mouse acceleration pre-applied
 
-static bool updated = false;       // true = mouse state update waits to be picked up
-static VMwareButtons buttons;      // state of mouse buttons, in VMware format
+static float pos_x = 0.0f; // absolute mouse position in guest-side pixels
+static float pos_y = 0.0f;
+
 static uint16_t scaled_x = 0x7fff; // absolute position scaled from 0 to 0xffff
 static uint16_t scaled_y = 0x7fff; // 0x7fff is a center position
-static int8_t counter_w  = 0;      // wheel movement counter
-
-static float pos_x = 0.0f;
-static float pos_y = 0.0f;
 
 // Multiply scale by 0.02f to put acceleration_vmm in a reasonable
 // range, similar to sensitivity_dos or sensitivity_vmm)
 constexpr float acceleration_multiplier = 0.02f;
 static MouseSpeedCalculator speed_xy(acceleration_multiplier *mouse_predefined.acceleration_vmm);
 
 // ***************************************************************************
-// VMware interface implementation
+// Common part implementation
 // ***************************************************************************
 
-static void MOUSEVMM_Activate()
+static void mouse_vmm_activate(const VmmProtocol protocol)
 {
-	if (!mouse_shared.active_vmm) {
-		mouse_shared.active_vmm = true;
+	bool is_activating    = false;
+	const bool was_active = mouse_shared.active_vmm;
+
+	if (protocol == VmmProtocol::VirtualBox && !virtualbox.is_active) {
```

Comment:
Yeah, so going by `VmmProtocol` here, maybe use `VmWare` capitalisation in the code instead of the rather funky `VMware`? That's a "marketing capitalisation"—we don't care about that 😄

Reply:
Changed capitalization in the source code to `VmWare` (a 'coding rules capitalization'). Comments are still using the 'marketing' capitalization, `VMware` - as this one is used since version 1.0 released in the late 90's.